examples/auctions/blind_auction.vy

```diff
@@ -0,0 +1,183 @@
+# Place a blinded bid with:
+# if fake:
+#     fakeInt = 1
+# else:
+#     fakeInt = 0
+# _blindedBid = sha3(concat(
+#       convert(value, bytes32),
+#       convert(fakeInt, bytes32),
+#       secret)
+# )
+#
+# The sent ether is only refunded if the bid is correctly revealed in the
+# revealing phase. The bid is valid if the ether sent together with the bid is
+# at least "value" and "fake" is not true. Setting "fake" to true and sending
+# not the exact amount are ways to hide the real bid but still make the
+# required deposit. The same address can place multiple bids.
+@public
+@payable
+def bid(_blindedBid: bytes32):
+    # Check if bidding period is still open
+    assert block.timestamp < self.biddingEnd
+
+    # Check that payer hasn't already placed maximum number of bids
+    # Note: because Vyper does not allow for dynamic arrays, we have limited the
+    # number of bids that can be placed by one address to 128 in this example
+    numBids: int128 = self.bidCounts[msg.sender]
+    assert numBids < 128
+
+    # Add bid to mapping of all bids
+    (self.bids[msg.sender])[numBids] = {blindedBid: _blindedBid, deposit: msg.value}
+    self.bidCounts[msg.sender] += 1
+
+
+# Returns a boolean value, `True` if bid placed successfully, `False` otherwise.
+@private
+def placeBid(bidder: address, value: wei_value) -> bool:
+    # If bid is less than highest bid, bid fails
+    if (value <= self.highestBid):
+        return False
+
+    # Refund the previously highest bidder
+    if (self.highestBidder != ZERO_ADDRESS):
+        self.pendingReturns[self.highestBidder] += self.highestBid
+
+    # Place bid successfully and update auction state
+    self.highestBid = value
+    self.highestBidder = bidder
+
+    return True
+
+
+# Reveal your blinded bids. You will get a refund for all correctly blinded
+# invalid bids and for all bids except for the totally highest.
+@public
+def reveal(_numBids: int128, _values: wei_value[128], _fakes: bool[128], _secrets: bytes32[128]):
+    # Check that bidding period is over
+    assert block.timestamp > self.biddingEnd
+
+    # Check that reveal end has not passed
+    assert block.timestamp < self.revealEnd
+
+    # Check that number of bids being revealed matches log for sender
+    assert _numBids == self.bidCounts[msg.sender]
+
+    # Calculate refund for sender
+    # Note: because Vyper does not allow for dynamic arrays, we have limited
+    # the number of bids that can be placed by one address to 128 in this example
+    refund: wei_value
+    for i in range(128):
+        # Note that loop may break sooner than 128 iterations if i >= _numBids
+        if (i >= _numBids):
+            break
+
+        # Get bid to check
+        bidToCheck: {blindedBid: bytes32, deposit: wei_value} = (self.bids[msg.sender])[i]
+
+        # Check against encoded packet
+        value: wei_value = _values[i]
+        fake: bool = _fakes[i]
+        # TODO: the block below can eventually be removed and boolean `fake`
```

Comment:
@fubuloubu @jacqueswww finished adding pretty comprehensive tests for this example, only outstanding thing is addressing this TODO, which I will do once #1077 is merged, but otherwise I think the example is ready for review 😄